crates/uv-trampoline/src/bounce.rs

```diff
@@ -37,32 +37,50 @@ fn getenv(name: &CStr) -> Option<CString> {
     }
 }
 
+/// Transform `<command> <arguments>` to `python <command> <arguments>`.
 fn make_child_cmdline(is_gui: bool) -> Vec<u8> {
-    unsafe {
-        let python_exe = find_python_exe(is_gui);
-
-        let my_cmdline = CStr::from_ptr(GetCommandLineA() as _);
-        let mut child_cmdline = Vec::<u8>::new();
-        child_cmdline.push(b'"');
-        for byte in python_exe.as_bytes() {
-            if *byte == b'"' {
-                // 3 double quotes: one to end the quoted span, one to become a literal double-quote,
-                // and one to start a new quoted span.
-                child_cmdline.extend(br#"""""#);
-            } else {
-                child_cmdline.push(*byte);
-            }
+    let executable_name: CString = executable_filename();
+    let python_exe = find_python_exe(is_gui, &executable_name);
+    let mut child_cmdline = Vec::<u8>::new();
+
+    push_quoted_path(&python_exe, &mut child_cmdline);
+    child_cmdline.push(b' ');
+
+    // Use the full executable name because CMD only passes the name of the executable (but not the path)
+    // when e.g. invoking `black` instead of `<PATH_TO_VENV>/Scripts/black` and Python then fails
+    // to find the file. Unfortunately, this complicates things because we now need to split the executable
+    // from the arguments string...
+    push_quoted_path(&executable_name, &mut child_cmdline);
+
+    push_arguments(&mut child_cmdline);
+    child_cmdline.push(b'\0');
+
+    eprintln!(
+        "executable_name: '{}'\nnew_cmdline: {}",
+        unsafe { core::str::from_utf8_unchecked(executable_name.to_bytes()) },
+        unsafe { core::str::from_utf8_unchecked(child_cmdline.as_slice()) }
```

Comment:
I don't think we need to use unchecked conversion here? The cost of UTF-8 validation on short strings is likely insignificant compared to starting a process.

Reply:
(I think this was one of my main complaints with `uv-trampoline`. Very excessive use of `unsafe`.)

Reply:
I'm also not even sure that these unchecked conversions are correct. The child path in particular is coming from a `&CStr`, and that is guaranteed to be valid UTF-8. Something higher level _might_ be guaranteeing it, but it is tricky for me to reason about in this context.

Reply:
These are debug statements, they'll go away ;)